pkg/reconciler/knative/resources/knative_configuration.go

```diff
@@ -74,14 +84,34 @@ func CreateKnativeConfiguration(name string, metadata metav1.ObjectMeta, modelSp
 							ServiceAccountName: modelSpec.ServiceAccountName,
 						},
 					},
-					Container: modelSpec.CreateModelServingContainer(metadata.Name),
+					Container: modelSpec.CreateModelServingContainer(metadata.Name, c.getFrameworkImage(modelSpec)),
 				},
 			},
 		},
 	}
 	return configuration
 }
 
+func (c *ConfigurationBuilder) getFrameworkImage(modelSpec *v1alpha1.ModelSpec) string {
+	if modelSpec.XGBoost != nil {
+		if imageName, ok := c.config.Data[v1alpha1.XGBoostServingImageConfigName]; ok {
+			return imageName
+		}
+		return v1alpha1.XGBoostServerImageName
+	} else if modelSpec.Tensorflow != nil {
```

Comment:
why not just pass the config into the handler and let it figure out what it needs? the whole point of the frameworks is to consolidate framework specific code and avoid these kinds of if blocks all over the place!

Reply:
We'd avoid passing the entire configmap, maybe as you suggested we can create a config per framework ?

Reply:
A config per framework makes sense to me - doesn't seem like a good idea to couple them... That might also allow us to have consistent names and maybe do some knative style config duck typing? ie imageName instead of xxxImageName... would allow us to have common code to apply common defaults.

Reply:
Added the config for each framework.